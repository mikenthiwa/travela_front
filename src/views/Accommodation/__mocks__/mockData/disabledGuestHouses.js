const disabledGuestHouses = [
  {
    id: 'juhiuhoji0',
    houseName: 'Guest House A',
    location: 'Nairobi, Kenya',
    bathRooms: 5,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'bvuh9iei8bh',
        roomName: 'ZAPTIU',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'NBiojnn98u',
        roomName: 'AFEIK',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'jbg8oijjolp',
    houseName: 'Guest House B',
    location: 'Nairobi, Kenya',
    bathRooms: 3,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'bnvjhijkpj',
        roomName: 'APLI',
        roomType: 'Single',
        bedCount: 1,
        faulty: false,
      },
      {
        id: 'bnvuiohj',
        roomName: 'Onij',
        roomType: 'Single',
        bedCount: 4,
        faulty: false,
      }
    ]
  },
  {
    id: 'nbuiojkpl',
    houseName: 'Guest House C',
    bathRooms: 4,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Lagos, Nigeria',
    rooms: [
      {
        id: 'bvuijkok',
        roomName: 'Two',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
    ]
  },
  {
    id: 'nmbkihjokko',
    houseName: 'Guest House D',
    bathRooms: 1,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Kigali, Rwanda',
    rooms: [
      {
        id: 'jbihjjkpk',
        roomName: 'Roof',
        roomType: 'Suite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
];

export default disabledGuestHouses;
