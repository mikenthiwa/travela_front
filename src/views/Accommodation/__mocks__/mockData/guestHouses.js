const guestHouses = [
  {
    id: 'wrtiuy56_u',
    houseName: 'Guest House 1',
    location: 'Nairobi, Kenya',
    bathRooms: 5,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'tryuompl0',
        roomName: 'Nyati',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'trytrimo',
        roomName: 'Maputi',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'wrtnytom_9',
    houseName: 'Guest House 2',
    location: 'Nairobi, Kenya',
    bathRooms: 3,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'plmyt0pol',
        roomName: 'Santana',
        roomType: 'Single',
        bedCount: 1,
        faulty: false,
      },
      {
        id: 'plmmnuyt',
        roomName: 'Shiloh',
        roomType: 'Single',
        bedCount: 4,
        faulty: false,
      }
    ]
  },
  {
    id: 'bytrwqjk9',
    houseName: 'Guest House 3',
    bathRooms: 4,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Lagos, Nigeria',
    rooms: [
      {
        id: 'btrfdseq23',
        roomName: 'Aben',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'btrfdsnytr3',
        roomName: 'Aben',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'requntm0',
    houseName: 'Guest House 4',
    bathRooms: 1,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Kigali, Rwanda',
    rooms: [
      {
        id: 'mnytbvfdc9',
        roomName: 'Berger',
        roomType: 'Suite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
];

export default guestHouses;
