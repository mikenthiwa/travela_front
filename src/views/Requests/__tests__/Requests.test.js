import React from 'react';
import sinon from 'sinon';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Requests, mapStateToProps } from '..';
import travelChecklistMockData from '../../../mockData/travelChecklistMockData';
import beds from '../../AvailableRooms/__mocks__/mockData/availableRooms';
import { submissionInfo } from '../../../mockData/checklistSubmissionMockData';

let props = {
  requests: [
    {
      id: 'xDh20btGz',
      name: 'Amarachukwo Agbo',
      tripType: 'oneWay',
      manager: 'Ezrqn Kiptanui',
      gender: 'Female',
      status: 'Open',
      trips: [
        {
          departureDate: '2018-09-20',
          origin: 'Lagos',
          destination: 'Angola',
          bedId: beds[0].id
        }
      ],
      department: 'TDD',
      role: 'Learning Facilitator'
    },
    {
      id: 'xDh20btGy',
      name: 'Amarachukwo Agbo',
      tripType: 'oneWay',
      manager: 'Ezrqn Kiptanui',
      gender: 'Female',
      status: 'Approved',
      trips: [
        {
          departureDate: '2018-09-20',
          origin: 'Lagos',
          destination: 'Angola',
          bedId: beds[0].id
        }
      ],
      department: 'TDD',
      role: 'Learning Facilitator'
    },
    {
      id: 'xDh20btGx',
      name: 'Amarachukwo Agbo',
      tripType: 'oneWay',
      manager: 'Ezrqn Kiptanui',
      gender: 'Female',
      status: 'Verified',
      trips: [
        {
          departureDate: '2018-09-20',
          origin: 'Lagos',
          destination: 'Angola',
          bedId: beds[0].id
        }
      ],
      department: 'TDD',
      role: 'Learning Facilitator'
    }
  ],
  requestOnEdit: {
    id: '1',
    name: 'Seun Undefined',
    tripType: 'multi',
    manager: 'Faluyi Seun',
    gender: 'Male',
    department: 'Talent & Development',
    role: 'Software Developer',
    status: 'Open',
    userId: 'lorem-ipsum',
    createdAt: '2018-09-26T15:15:49.808Z',
    updatedAt: '2018-09-26T15:15:49.808Z',
    trips: [
      {
        id: '1',
        origin: 'Abuja Nigeria',
        destination: 'Lagos Nigeria',
        departureDate: '2018-09-30',
        returnDate: '2018-09-30',
        createdAt: '2018-09-27T18:49:03.626Z',
        updatedAt: '2018-09-27T18:49:43.803Z',
        requestId: 'NfR-9KoCP',
        bedId: beds[0].id
      }
    ]
  },
  availableRooms: {
    beds
  },
  userData: {
    result: {
      id: '2',
      fullName: 'Collins Muru',
      email: 'collins.muru@andela.com',
      userId: '-LJNw1BsT0LP_E4l2peP',
      passportName: 'Collins Njau',
      department: 'Talent & Development',
      occupation: 'Software Developer',
      manager: 'Collins',
      gender: 'Male',
      createdAt: '2018-09-14T12:48:11.266Z',
      updatedAt: '2018-09-16T07:53:48.835Z',
      roleId: 401938
    }
  },
  pagination: {
    currentPage: 1,
    pageCount: 4,
    dataCount: 10,
    onPageChange: sinon.spy()
  },
  fetchEditRequest: sinon.spy(() => Promise.resolve()),
  fetchUserRequests: sinon.spy(() => Promise.resolve()),
  fetchRoleUsers: sinon.spy(() => Promise.resolve()),
  updateUserProfile: sinon.spy(() => Promise.resolve()),
  fetchAvailableRooms: sinon.spy(() => Promise.resolve()),
  fetchTravelChecklist: sinon.spy(() => Promise.resolve()),
  deleteRequest: sinon.spy(() => Promise.resolve()),
  fetchUserRequestsError: null,
  openRequestsCount: 1,
  pastRequestsCount: 1,
  isLoading: false,
  history: {
    push: jest.fn()
  },
  location: {
    search: '?page=1',
    pathname: '/requests'
  },
  user: {
    UserInfo: {
      name: 'John Doe'
    }
  },
  createNewRequest: jest.fn(),
  getOccupation: jest.fn(),
  getUserData: jest.fn(),
  fetchAvailableRoomsSuccess: jest.fn(),
  fetchAllTravelStipends: jest.fn(),
  loading: false,
  errors: [],
  shouldOpen: false,
  fetchCenters: jest.fn(),
  requestStatus: 'Open',
  modalType: null,
  openModal: sinon.spy(() => Promise.resolve()),
  closeModal: sinon.spy(() => Promise.resolve()),
  page: 'Requests',
  match: {
    params: { requestId: 'xDh20btGx' }
  },
  editRequest: jest.fn(),
  submissionInfo,
  fileUploads: {},
  fetchSubmission: jest.fn(),
  postSubmission: jest.fn(),
  validateTrips: jest.fn(),
};

const initialState = {
  auth: {
    isAuthenticated: true,
    user: {
      UserInfo: {
        name: 'Tomato Jos',
        picture: 'http://picture.com/gif'
      }
    }
  },
  user: { userData: { result: {} } },
  requestsReducer: {
    requests: [],
    request: {},
    loading: false,
    errors: []
  },
  modalReducer: {
    shouldOpen: false,
    modalType: null
  },
  roleUsers: [
    { fullName: 'Samuel Kubai', email: 'samuel@andela.com' },
    { fullName: 'Chris Akanmu', email: 'chris@andela.com' }
  ],
  getCurrentUserRole: 'tomato',
  travelChecklist: { checklistItems: travelChecklistMockData }
};
const mockStore = configureStore();
const store = mockStore(initialState);

describe('<Requests>', () => {
  process.env.REACT_APP_CITY = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyD-fvLImnNbTfYV3Pd1nJuK7NbzZJNr4ug&libraries=places';

  it('should render the Requests without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.length).toBe(1);
    wrapper.unmount();
  });

  it('calls the componentDidMount method', () => {
    const spy = sinon.spy(Requests.prototype, 'componentDidMount');
    const { fetchUserRequests, fetchRoleUsers, requestId, openModal} = props;
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(spy.called).toEqual(true);
    expect(fetchUserRequests.called).toEqual(true);
    expect(fetchUserRequests.calledWith('?page=1')).toEqual(true);
    expect(fetchRoleUsers.called).toEqual(true);
    expect(openModal.called).toEqual(true);
    expect(fetchRoleUsers.calledWith(53019)).toEqual(true);
    wrapper.unmount();
  });


  it(`calls the fetchRequests method
    when the next pagination button is clicked`, () => {
    const { fetchUserRequests } = props;
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests {...props} />
        </MemoryRouter>
      </Provider>
    );
    const spy = sinon.spy(wrapper.find(Requests).instance(), 'fetchRequests');
    wrapper.find('#next-button').simulate('click');
    expect(spy.calledOnce).toEqual(true);
    expect(fetchUserRequests.called).toEqual(true);
    wrapper.unmount();
  });

  it(`calls the fetchRequests method with the correct query
    when the next pagination button is clicked`, () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests
            {...{
              ...props,
              location: {
                search: ''
              }
            }}
          />
        </MemoryRouter>
      </Provider>
    );
    const spy = sinon.spy(wrapper.find(Requests).instance(), 'fetchRequests');
    wrapper.find('#next-button').simulate('click');
    expect(spy.called).toEqual(true);
    expect(spy.calledWith('?page=2')).toEqual(true);
    wrapper.unmount();
  });

  it(`calls the fetchRequests method
    with the selected limit when a limit is selected`, () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests {...props} />
        </MemoryRouter>
      </Provider>
    );
    const spy = sinon.spy(wrapper.find(Requests).instance(), 'fetchRequests');
    wrapper
      .find('.dropdown__list__item')
      .first()
      .simulate('click');
    expect(spy.calledOnce).toEqual(true);
    expect(spy.calledWith('?page=1&limit=10')).toEqual(true);
    wrapper.unmount();
  });

  it(`calls the fetchRequests method with the last possible page
  when a limit and a page above the available pages is selected`, () => {
    const { pagination } = props;
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests
            {...{
              ...props,
              pagination: { ...pagination, currentPage: 4 },
              location: { search: '?page=4&status=open' }
            }}
          />
        </MemoryRouter>
      </Provider>
    );
    const spy = sinon.spy(wrapper.find(Requests).instance(), 'fetchRequests');
    wrapper
      .find('.dropdown__list__item')
      .last()
      .simulate('click');
    expect(spy.calledOnce).toEqual(true);
    expect(spy.calledWith('?page=1&status=open&limit=30')).toEqual(true);
    wrapper.unmount();
  });

  it('should render the child components', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find('.rp-requests__header').length).toBe(1); // RequestsPanelHeader
    expect(wrapper.find('Table').length).toBe(1);
    expect(wrapper.find('RequestPanelHeader').length).toBe(1);
    expect(wrapper.find('WithLoading').length).toBe(1); //WithLoading HOC containing Requests
    // Since the element is always on the DOM, the page length will always be one
    expect(wrapper.find('Pagination').length).toBe(1);
    wrapper.unmount();
  });

  it('should call renderRequestPanelHeader method', () => {
    const wrapper = shallow(<Requests {...props} />);
    const instance = wrapper.instance();
    const spyon = jest.spyOn(instance, 'renderRequestPanelHeader');
    expect(spyon).toHaveBeenCalledTimes(0);
  });

  it('renders as expected', () => {
    const wrapper = shallow(<Requests {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should set `shouldOpen` prop to `true` when new request button is clicked', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests {...{...props, shouldOpen: true, modalType: 'new model'}} />
        </MemoryRouter>
      </Provider>
    );
    wrapper.find('.btn-new-request').simulate('click');
    expect(wrapper.find('Requests').props().shouldOpen).toBe(true);
    expect(props.openModal.called).toBe(true);
  });

  it('should set `visibility` prop to `visible` when new request button is clicked', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Requests
            {...{ ...props, shouldOpen: true, modalType: 'new model' }}
          />
        </MemoryRouter>
      </Provider>
    );
    wrapper.find('.btn-new-request').simulate('click');
    expect(
      wrapper
        .find('Modal')
        .at(0)
        .props().visibility
    ).toEqual('visible');
  });

  it('maps state to props and return the expected object', () => {
    const requests = {
      requests: [],
      errors: [],
      loading: false
    };

    const modal = {
      modal: {
        shouldOpen: false,
        modalType: null
      }
    };
    const user = {
      getUserData: {}
    };

    const travelChecklist = {
      checklistItems: travelChecklistMockData,
      isLoading: false
    };
    const props = mapStateToProps({requests, modal, user, travelChecklist});
    expect(props).toEqual({
      ...requests, ...modal.modal,
      userData: user.getUserData,
      travelChecklists: travelChecklist
    });
  });


  it('should handle edit request', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Requests {...{...props}} />
      </Provider>
    );
    const mountWrapper = wrapper.find('li#iconBtn');
    mountWrapper.simulate('click');
    const { history: { push } } = props;
    expect(push).toBeCalledWith('/requests/edit-request/xDh20btGz');
  });

  it('should handle delete request', () => {
    const event = { preventDefault: jest.fn() };
    const newProps = {
      ...props,
      shouldOpen: true,
      modalType: 'delete request'
    };
    const wrapper = shallow(<Requests {...newProps} />
    );
    const instance = wrapper.instance();
    const spyon = jest.spyOn(instance, 'handleDeleteRequest');
    wrapper.instance().handleDeleteRequest('xDh20btGz');
    expect(spyon).toHaveBeenCalledTimes(1);
    expect(instance.props.deleteRequest.called).toBe(true);
    expect(instance.props.deleteRequest.calledWith('xDh20btGz')).toBe(true);
  });

  it('should handle close travel checklist submission modal', () => {
    const wrapper = shallow(<Requests {...props} />
    );
    const instance = wrapper.instance();
    const handleCloseSubmissionModalSpy = jest.spyOn(instance, 'handleCloseSubmissionModal');
    wrapper.instance().handleCloseSubmissionModal();
    expect(handleCloseSubmissionModalSpy).toHaveBeenCalledTimes(1);
  });

  it('should handle close travel checklist modal', () => {
    const wrapper = shallow(<Requests {...props} />
    );
    const instance = wrapper.instance();
    const handleCloseChecklistModalSpy = jest.spyOn(instance, 'handleCloseChecklistsModal');
    wrapper.instance().handleCloseChecklistsModal();
    expect(handleCloseChecklistModalSpy).toHaveBeenCalledTimes(1);
  });

  it('should set openChecklist to true', () => {
    const prevProps = {
      ...props,
      match: {
        path: '/requests/:requestId/checklist',
        params: {requestId: 'xDh20btGz'}
      },
      requests: []
    };
    const wrapper = mount(<Requests {...prevProps} />);
    wrapper.setProps({...props});
    expect(wrapper.state('openChecklist')).toBe(true);
  });

});
