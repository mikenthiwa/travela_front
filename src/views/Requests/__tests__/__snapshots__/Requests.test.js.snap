// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Requests> renders as expected 1`] = `
<Fragment>
  <Modal
    closeDeleteModal={null}
    closeModal={[Function]}
    customModalStyles=""
    customOverlayStyle=""
    modalBar={<div />}
    modalContentId=""
    modalId=""
    requestId=""
    showOverlay={true}
    title="New Travel Request"
    visibility="invisible"
    width="81.95%"
  >
    <NewRequestForm
      availableRooms={
        Object {
          "beds": Array [
            Object {
              "bedName": "bed 1",
              "booked": false,
              "createdAt": "2018-08-16T11:11:52.181Z",
              "id": 1,
              "roomId": "room-id-1",
              "rooms": Object {
                "bedCount": 2,
                "createdAt": "2018-08-16T11:11:52.181Z",
                "faulty": false,
                "guestHouseId": "guest-house-1",
                "guestHouses": Object {
                  "bathRooms": 1,
                  "createdAt": "2018-08-16T11:11:52.181Z",
                  "genderPolicy": "unisex",
                  "houseName": "Mini flat-B",
                  "id": "guest-house-1",
                  "imageUrl": "https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1",
                  "location": "Lagos,Nigeria",
                  "updatedAt": "2018-08-16T11:11:52.181Z",
                  "userId": "-LJV4b1QTCYewOtk5F63",
                },
                "id": "room-id-1",
                "roomName": "Kwetu",
                "roomType": "ensuited",
                "updatedAt": "2018-08-16T11:11:52.181Z",
              },
              "updatedAt": "2018-08-16T11:11:52.181Z",
            },
            Object {
              "bedName": "bed 2",
              "booked": false,
              "createdAt": "2018-08-16T11:11:52.181Z",
              "id": 2,
              "roomId": "room-id-1",
              "rooms": Object {
                "bedCount": 2,
                "createdAt": "2018-08-16T11:11:52.181Z",
                "faulty": false,
                "guestHouseId": "guest-house-1",
                "guestHouses": Object {
                  "bathRooms": 1,
                  "createdAt": "2018-08-16T11:11:52.181Z",
                  "genderPolicy": "unisex",
                  "houseName": "Mini flat-B",
                  "id": "guest-house-1",
                  "imageUrl": "https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1",
                  "location": "Lagos,Nigeria",
                  "updatedAt": "2018-08-16T11:11:52.181Z",
                  "userId": "-LJV4b1QTCYewOtk5F63",
                },
                "id": "room-id-1",
                "roomName": "Kwetu",
                "roomType": "ensuited",
                "updatedAt": "2018-08-16T11:11:52.181Z",
              },
              "updatedAt": "2018-08-16T11:11:52.181Z",
            },
            Object {
              "bedName": "bed 1",
              "booked": false,
              "createdAt": "2018-08-16T11:11:52.181Z",
              "id": 3,
              "roomId": "room-id-2",
              "rooms": Object {
                "bedCount": 3,
                "createdAt": "2018-08-16T11:11:52.181Z",
                "faulty": false,
                "guestHouseId": "guest-house-1",
                "guestHouses": Object {
                  "bathRooms": 1,
                  "createdAt": "2018-08-16T11:11:52.181Z",
                  "genderPolicy": "unisex",
                  "houseName": "Mini flat-B",
                  "id": "guest-house-1",
                  "imageUrl": "https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1",
                  "location": "Lagos,Nigeria",
                  "updatedAt": "2018-08-16T11:11:52.181Z",
                  "userId": "-LJV4b1QTCYewOtk5F63",
                },
                "id": "room-id-2",
                "roomName": "Menengai",
                "roomType": "ensuited",
                "updatedAt": "2018-08-16T11:11:52.181Z",
              },
              "updatedAt": "2018-08-16T11:11:52.181Z",
            },
            Object {
              "bedName": "bed 2",
              "booked": false,
              "createdAt": "2018-08-16T11:11:52.181Z",
              "id": 4,
              "roomId": "room-id-2",
              "rooms": Object {
                "bedCount": 3,
                "createdAt": "2018-08-16T11:11:52.181Z",
                "faulty": false,
                "guestHouseId": "guest-house-1",
                "guestHouses": Object {
                  "bathRooms": 1,
                  "createdAt": "2018-08-16T11:11:52.181Z",
                  "genderPolicy": "unisex",
                  "houseName": "Mini flat-B",
                  "id": "guest-house-1",
                  "imageUrl": "https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1",
                  "location": "Lagos,Nigeria",
                  "updatedAt": "2018-08-16T11:11:52.181Z",
                  "userId": "-LJV4b1QTCYewOtk5F63",
                },
                "id": "room-id-2",
                "roomName": "Menengai",
                "roomType": "ensuited",
                "updatedAt": "2018-08-16T11:11:52.181Z",
              },
              "updatedAt": "2018-08-16T11:11:52.181Z",
            },
            Object {
              "bedName": "bed 3",
              "booked": false,
              "createdAt": "2018-08-16T11:11:52.181Z",
              "id": 5,
              "roomId": "room-id-2",
              "rooms": Object {
                "bedCount": 3,
                "createdAt": "2018-08-16T11:11:52.181Z",
                "faulty": false,
                "guestHouseId": "guest-house-1",
                "guestHouses": Object {
                  "bathRooms": 1,
                  "createdAt": "2018-08-16T11:11:52.181Z",
                  "genderPolicy": "unisex",
                  "houseName": "Mini flat-B",
                  "id": "guest-house-1",
                  "imageUrl": "https://www.dropbox.com/s/c2n63x2jamh3ndv/guesthouse2.jpg?raw=1",
                  "location": "Lagos,Nigeria",
                  "updatedAt": "2018-08-16T11:11:52.181Z",
                  "userId": "-LJV4b1QTCYewOtk5F63",
                },
                "id": "room-id-2",
                "roomName": "Menengai",
                "roomType": "ensuited",
                "updatedAt": "2018-08-16T11:11:52.181Z",
              },
              "updatedAt": "2018-08-16T11:11:52.181Z",
            },
          ],
        }
      }
      closeModal={[Function]}
      creatingRequest={false}
      editing={false}
      errors={Array []}
      fetchAllTravelStipends={[Function]}
      fetchAvailableRooms={[Function]}
      fetchAvailableRoomsSuccess={[Function]}
      fetchTravelCostsByLocation={[Function]}
      fetchUserRequests={[Function]}
      handleCreateRequest={[Function]}
      handleEditRequest={[Function]}
      history={Object {}}
      loading={false}
      managers={Array []}
      modalType={null}
      requestOnEdit={
        Object {
          "createdAt": "2018-09-26T15:15:49.808Z",
          "department": "Talent & Development",
          "gender": "Male",
          "id": "1",
          "location": undefined,
          "manager": "Faluyi Seun",
          "name": "Seun Undefined",
          "role": "Software Developer",
          "status": "Open",
          "tripType": "multi",
          "trips": Array [
            Object {
              "bedId": 1,
              "createdAt": "2018-09-27T18:49:03.626Z",
              "departureDate": "2018-09-30",
              "destination": "Lagos Nigeria",
              "id": "1",
              "origin": "Abuja Nigeria",
              "requestId": "NfR-9KoCP",
              "returnDate": "2018-09-30",
              "updatedAt": "2018-09-27T18:49:43.803Z",
            },
          ],
          "updatedAt": "2018-09-26T15:15:49.808Z",
          "userId": "lorem-ipsum",
        }
      }
      updateUserProfile={[Function]}
      user={
        Object {
          "UserInfo": Object {
            "name": "John Doe",
          },
        }
      }
      userData={
        Object {
          "createdAt": "2018-09-14T12:48:11.266Z",
          "department": "Talent & Development",
          "email": "collins.muru@andela.com",
          "fullName": "Collins Muru",
          "gender": "Male",
          "id": "2",
          "manager": "Collins",
          "occupation": "Software Developer",
          "passportName": "Collins Njau",
          "roleId": 401938,
          "updatedAt": "2018-09-16T07:53:48.835Z",
          "userId": "-LJNw1BsT0LP_E4l2peP",
        }
      }
      validateTrips={[Function]}
    />
  </Modal>
  <div
    className="rp-requests__header"
  >
    <RequestPanelHeader
      activeStatus="all"
      fetchRequests={[Function]}
      getRequestsWithLimit={[Function]}
      loading={false}
      modalType={null}
      openModal={[Function]}
      openNewRequestPage={[Function]}
      openRequestsCount={1}
      pastRequestsCount={1}
      requestsLength={3}
      shouldOpen={false}
      url="?page=1"
    />
  </div>
  <div
    className="rp-table"
  >
    <WithLoading
      closeModal={[Function]}
      deleteRequest={[Function]}
      editRequest={[Function]}
      fetchRequestsError={null}
      fetchSubmission={[Function]}
      fetchUserRequests={[Function]}
      fileUploads={Object {}}
      handleCloseChecklistModal={[Function]}
      handleCloseSubmissionModal={[Function]}
      history={
        Object {
          "push": [Function],
        }
      }
      isFetching={false}
      isLoading={false}
      location={
        Object {
          "pathname": "/requests",
          "search": "?page=1",
        }
      }
      modalType={null}
      openChecklist={false}
      openModal={[Function]}
      page="Requests"
      postSubmission={[Function]}
      requestId="xDh20btGx"
      requests={
        Array [
          Object {
            "department": "TDD",
            "gender": "Female",
            "id": "xDh20btGz",
            "manager": "Ezrqn Kiptanui",
            "name": "Amarachukwo Agbo",
            "role": "Learning Facilitator",
            "status": "Open",
            "tripType": "oneWay",
            "trips": Array [
              Object {
                "bedId": 1,
                "departureDate": "2018-09-20",
                "destination": "Angola",
                "origin": "Lagos",
              },
            ],
          },
          Object {
            "department": "TDD",
            "gender": "Female",
            "id": "xDh20btGy",
            "manager": "Ezrqn Kiptanui",
            "name": "Amarachukwo Agbo",
            "role": "Learning Facilitator",
            "status": "Approved",
            "tripType": "oneWay",
            "trips": Array [
              Object {
                "bedId": 1,
                "departureDate": "2018-09-20",
                "destination": "Angola",
                "origin": "Lagos",
              },
            ],
          },
          Object {
            "department": "TDD",
            "gender": "Female",
            "id": "xDh20btGx",
            "manager": "Ezrqn Kiptanui",
            "name": "Amarachukwo Agbo",
            "role": "Learning Facilitator",
            "status": "Verified",
            "tripType": "oneWay",
            "trips": Array [
              Object {
                "bedId": 1,
                "departureDate": "2018-09-20",
                "destination": "Angola",
                "origin": "Lagos",
              },
            ],
          },
        ]
      }
      setOpenChecklist={[Function]}
      shouldOpen={false}
      showTravelChecklist={[Function]}
      submissionInfo={
        Object {
          "fetchFailureMessage": "",
          "fetchSuccessMessage": "",
          "isFetching": false,
          "isLoading": false,
          "isUploading": Array [],
          "itemsToCheck": Array [
            "NumC5-pK7G-9",
            "NumC5-pK7G-8",
            "H5X8CVA0IR-9",
            "NumC5-pK7G-3",
          ],
          "percentageCompleted": 0,
          "postFail": false,
          "postSuccess": Array [],
          "requestId": "",
          "submissions": Array [
            Object {
              "checklist": Array [
                Object {
                  "deleteReason": null,
                  "destinationName": "Lagos, Nigeria",
                  "id": "9",
                  "name": "yellow card",
                  "requiresFiles": false,
                  "resources": Array [],
                  "submissions": Array [
                    Object {
                      "checklistItemId": "9",
                      "checklistSubmissions": Object {
                        "id": "9",
                      },
                      "createdAt": "2018-11-18T22:03:26.442Z",
                      "deletedAt": null,
                      "id": "JGLqrmBEd",
                      "tripId": "NumC5-pK7G",
                      "updatedAt": "2018-11-18T23:53:59.297Z",
                      "value": "I submitted yellow card",
                    },
                  ],
                },
                Object {
                  "deleteReason": null,
                  "destinationName": "Lagos, Nigeria",
                  "id": "8",
                  "name": "visa application",
                  "requiresFiles": true,
                  "resources": Array [],
                  "submissions": Array [
                    Object {
                      "checklistItemId": "8",
                      "checklistSubmissions": Object {
                        "id": "8",
                      },
                      "createdAt": "2018-11-18T22:03:50.071Z",
                      "deletedAt": null,
                      "id": "wY4s9r-29",
                      "tripId": "NumC5-pK7G",
                      "updatedAt": "2018-11-18T22:26:25.322Z",
                      "value": Object {
                        "fileName": "test.pdf",
                        "url": "image.com",
                      },
                    },
                  ],
                },
                Object {
                  "deleteReason": null,
                  "destinationName": "Lagos, Nigeria",
                  "id": "7",
                  "name": "Travel stipend",
                  "requiresFiles": true,
                  "resources": Array [],
                  "submissions": Array [
                    Object {
                      "checklistItemId": 9,
                      "checklistSubmissions": Object {
                        "id": 9,
                      },
                      "createdAt": "2019-03-21T05:03:51.398Z",
                      "deletedAt": null,
                      "documentId": null,
                      "id": "fopl4zG0p",
                      "tripId": "H5X8CVA0IR",
                      "updatedAt": "2019-03-21T05:03:51.398Z",
                      "value": Object {
                        "fileName": "test.png",
                        "url": "image.com",
                      },
                    },
                  ],
                },
                Object {
                  "deleteReason": null,
                  "destinationName": "Default",
                  "id": "3",
                  "name": "Travel Ticket Details",
                  "requiresFiles": false,
                  "resources": Array [],
                  "submissions": Array [
                    Object {
                      "checklistItemId": "3",
                      "checklistSubmissions": Object {
                        "id": "3",
                      },
                      "createdAt": "2018-11-18T22:04:32.606Z",
                      "deletedAt": null,
                      "id": "yRCCSp0Oy",
                      "tripId": "NumC5-pK7G",
                      "updatedAt": "2018-11-18T22:04:32.606Z",
                      "value": Object {
                        "airline": "Arik Airways",
                        "arrivalTime": "2018-11-20T01:00",
                        "departureTime": "2018-11-20T01:00",
                        "returnAirline": "",
                        "returnDepartureTime": "2018-11-20T01:00",
                        "returnTicketNumber": "LG 555",
                        "returnTime": "2018-11-20T01:00",
                        "ticketNumber": "LG 777",
                      },
                    },
                  ],
                },
                Object {
                  "deleteReason": null,
                  "destinationName": "Lagos, Nigeria",
                  "id": "10",
                  "name": "Travel stipend",
                  "requiresFiles": true,
                  "resources": Array [],
                  "submissions": Array [],
                },
              ],
              "destinationName": "Lagos, Nigeria",
              "tripId": "NumC5-pK7G",
            },
          ],
          "successMessage": "",
          "successStatus": false,
          "tripType": "",
        }
      }
      type="requests"
    />
  </div>
  <div
    className="rp-pagination"
  >
    <Pagination
      currentPage={1}
      onPageChange={[Function]}
      pageCount={4}
    />
  </div>
</Fragment>
`;
